jwst/dark_current/tests/test_dark_sub.py

```diff
@@ -0,0 +1,517 @@
+"""
+Unit tests for dark current correction
+"""
+
+import pytest
+import numpy as np
+
+from jwst.dark_current.dark_sub import average_dark_frames
+from jwst.datamodels import (
+    RampModel, 
+    DarkModel, 
+    MIRIRampModel, 
+    DarkMIRIModel, 
+    dqflags,
+)
+from jwst.dark_current.dark_sub import do_correction as darkcorr
+from jwst.dark_current import DarkCurrentStep
+from astropy.io import fits
+
+# Dictionary of NIRCam readout patterns
+DEEP8 = {}
+DEEP8['ngroups'] = 20.
+DEEP8['nframes'] = 8
+DEEP8['nskip'] = 12
+
+DEEP2 = {}
+DEEP2['ngroups'] = 20.
+DEEP2['nframes'] = 2
+DEEP2['nskip'] = 18
+
+MEDIUM8 = {}
+MEDIUM8['ngroups'] = 10.
+MEDIUM8['nframes'] = 8
+MEDIUM8['nskip'] = 2
+
+MEDIUM2 = {}
+MEDIUM2['ngroups'] = 10.
+MEDIUM2['nframes'] = 2
+MEDIUM2['nskip'] = 8
+
+SHALLOW4 = {}
+SHALLOW4['ngroups'] = 10.
+SHALLOW4['nframes'] = 4
+SHALLOW4['nskip'] = 1
+
+SHALLOW2 = {}
+SHALLOW2['ngroups'] = 10.
+SHALLOW2['nframes'] = 2
+SHALLOW2['nskip'] = 3
+
+BRIGHT2 = {}
+BRIGHT2['ngroups'] = 10.
+BRIGHT2['nframes'] = 2
+BRIGHT2['nskip'] = 0
+
+BRIGHT1 = {}
+BRIGHT1['ngroups'] = 10.
+BRIGHT1['nframes'] = 1
+BRIGHT1['nskip'] = 1
+
+RAPID = {}
+RAPID['ngroups'] = 10.
+RAPID['nframes'] = 1
+RAPID['nskip'] = 0
+
+READPATTERNS = {}
+READPATTERNS['DEEP8'] = DEEP8
+READPATTERNS['DEEP2'] = DEEP2
+READPATTERNS['MEDIUM8'] = MEDIUM8
+READPATTERNS['MEDIUM2'] = MEDIUM2
+READPATTERNS['SHALLOW4'] = SHALLOW4
+READPATTERNS['SHALLOW2'] = SHALLOW2
+READPATTERNS['BRIGHT2'] = BRIGHT2
+READPATTERNS['BRIGHT1'] = BRIGHT1
+READPATTERNS['RAPID'] = RAPID
+
+TFRAME = 10.73677
+
+
+def test_frame_averaging(setup_nrc_cube):
+    '''Check that if nframes>1 or groupgap>0, then the pipeline reconstructs
+       the dark reference file to match the frame averaging and groupgap
+       settings of the exposure.'''
+
+    # Values to build the fake data arrays
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
```

Comment:
Currently this is the test that takes 70% of the time in running these 8 tests, so it would be good to get this to use a bit less memory and go faster.

I think these sizes can be reduced dramatically so as to use less memory and speed the tests up.  Really just a 200x200 array with a few groups should be ok, right?  And instead of accessing pixel location `500,500` below, access `100,100`.